pallets/asset/src/lib.rs

```diff
@@ -2080,45 +1732,59 @@ impl<T: Trait> Module<T> {
         ticker: &Ticker,
         value: T::Balance,
     ) -> StdResult<u8, &'static str> {
-        Ok(if !Self::check_granularity(&ticker, value) {
-            // Granularity check
-            INVALID_GRANULARITY
-        } else if from_portfolio.did == to_portfolio.did {
-            INVALID_RECEIVER_DID
-        } else if !Identity::<T>::has_valid_cdd(from_portfolio.did) {
+        ensure_as_ok!(Self::check_granularity(&ticker, value), INVALID_GRANULARITY);
+        ensure_as_ok!(from_portfolio.did != to_portfolio.did, INVALID_RECEIVER_DID);
```

Comment:
The macro is cool but I would still prefer keeping all `if x { return Ok(y); }` as being universally readable and not requiring any macro knowledge.

Reply:
(`else if`s in this case since that's how the code is written right now and that seems simplest/DRYest)